crypto/bls12381_utils.c

```diff
@@ -583,28 +583,39 @@ void ep2_subtract_vector(ep2_t res, ep2_t x, ep2_st* y, int len){
     ep2_add_projc(res, x, res);
 }
 
+// computes the sum of the G1 array elements y and writes the sum in jointy
+void ep_sum_vector(ep_t jointx, ep_st* x, const int len) {
+    ep_set_infty(jointx);
+    for (int i=0; i<len; i++){
+        ep_add_projc(jointx, jointx, &x[i]);
+    }
+}
+
 // Computes the sum of the signatures (G1 elements) flattened in an single sigs array
 // and store the sum bytes in dest
 // The function assmues sigs is correctly allocated with regards to len.
-int ep_sum_vector(byte* dest, const byte* sigs, const int len) {
+int ep_sum_vector_byte(byte* dest, const byte* sigs_bytes, const int len) {
     // temp variables
-    ep_t acc, sig;        
+    ep_t acc;        
     ep_new(acc);
-    ep_new(sig);
     ep_set_infty(acc);
+    ep_st* sigs = (ep_st*) malloc(len * sizeof(ep_st));
 
-    // sum the points
+    // convert the points
     for (int i=0; i < len; i++) {
-        if (ep_read_bin_compact(sig, &sigs[SIGNATURE_LEN*i], SIGNATURE_LEN) != RLC_OK)
+        ep_new(sigs[i]);
+        if (ep_read_bin_compact(&sigs[i], &sigs_bytes[SIGNATURE_LEN*i], SIGNATURE_LEN) != RLC_OK)
             return INVALID;
-        ep_add_projc(acc, acc, sig);
     }
+    // sum the points
+    ep_sum_vector(acc, sigs, len);
```

Comment:
A comment about `ep_sum_vector`'s doc, 
`// The function assmues (-->assumes) sigs is correctly allocated with regards to len.`